python/cudf/cudf/core/tools/datetimes.py

```diff
@@ -459,30 +440,120 @@ def __init__(self, n=1, normalize=False, **kwds):
+            "days",
+            "hours",
+            "minutes",
+            "seconds",
+            "milliseconds",
+            "microseconds",
+            "nanoseconds",
         }
 
-        supported_kwargs = {"months"}
+        unsupported_units = all_possible_units - supported_units
+
+        invalid_kwds = set(kwds) - supported_units - unsupported_units
+        if invalid_kwds:
+            raise TypeError(
+                f"Keyword arguments '{','.join(list(invalid_kwds))}'"
+                " are not recognized"
+            )
+
+        unsupported_kwds = set(kwds) & unsupported_units
+        if unsupported_kwds:
+            raise NotImplementedError(
+                f"Keyword arguments '{','.join(list(unsupported_kwds))}'"
+                " are not yet supported."
+            )
+
+        if any(val != int(val) for val in kwds.values()):
+            raise ValueError("Non-integer periods not supported")
+
+        self._kwds = kwds
+        kwds = self._combine_months_and_years(**kwds)
+        kwds = self._combine_kwargs_to_seconds(**kwds)
 
         scalars = {}
         for k, v in kwds.items():
-            if k in all_possible_kwargs:
+            if k in all_possible_units:
                 # Months must be int16
-                dtype = "int16" if k == "months" else None
+                if k == "months":
+                    # TODO: throw for oob int16 vals
+                    dtype = "int16"
+                else:
+                    unit = self._UNITS_TO_CODES[k]
+                    dtype = np.dtype(f"timedelta64[{unit}]")
                 scalars[k] = cudf.Scalar(v, dtype=dtype)
 
-        super().__init__(n=n, normalize=normalize, **kwds)
+        self._scalars = scalars
+
+    @property
+    def kwds(self):
+        return self._kwds
+
+    def _combine_months_and_years(self, **kwargs):
+        # TODO: if months is zero, don't do a binop
+        kwargs["months"] = kwargs.pop("years", 0) * 12 + kwargs.pop(
+            "months", 0
+        )
+        return kwargs
 
-        wrong_kwargs = set(kwds.keys()).difference(supported_kwargs)
-        if len(wrong_kwargs) > 0:
-            raise ValueError(
-                f"Keyword arguments '{','.join(list(wrong_kwargs))}'"
-                " are not yet supported in cuDF DateOffsets"
+    def _combine_kwargs_to_seconds(self, **kwargs):
+        """
+        Combine days, weeks, hours and minutes to a single
+        scalar representing the total seconds
+        """
+        seconds = 0
+        seconds += kwargs.pop("weeks", 0) * 604800
+        seconds += kwargs.pop("days", 0) * 86400
+        seconds += kwargs.pop("hours", 0) * 3600
+        seconds += kwargs.pop("minutes", 0) * 60
+        seconds += kwargs.pop("seconds", 0)
+
+        if seconds > np.iinfo("int64").max:
+            raise OverflowError(
+                "Total days + weeks + hours + minutes + seconds can not exceed"
+                f" {np.iinfo('int64').max} seconds"
+            )
```

Comment:
Does Pandas have this behavior as well? I'm wondering if this should be a `NotImplementedError` instead.

Reply:
It doesn't; because `seconds` is the largest resolution libcudf supports, we convert the larger units to seconds. This can lead to us overflowing before Pandas. I agree that `NotImplementedError` would be more appropriate here. Fixed.